client/src/components/Panels/utilities.js

```diff
@@ -1,33 +1,96 @@
-export function filterToolSections(layout, results) {
-    if (results) {
-        results = normalize_results(results);
-        const filteredLayout = layout.map((section) => {
-            var toolRes = [];
-            if (section.elems) {
-                section.elems.forEach((el) => {
-                    if (
-                        (!el.text && results.includes(el.id)) ||
-                        (el.tool_shed_repository && results.includes(el.tool_shed_repository.name))
-                    ) {
-                        toolRes.push(el);
-                    }
-                });
-            }
-            // sort tools in section by rank in results
-            toolRes.sort((tool1, tool2) => {
-                return results.indexOf(tool1.id) - results.indexOf(tool2.id);
+import { orderBy } from "lodash";
+
+export class toolSearch {
```

Comment:
As mentioned, I suggest to not use a class instance here but just utility functions with defaults. These functions can be stateless.